libmamba/src/solv-cpp/repo.cpp

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) 2023, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#include <cassert>
+#include <cerrno>
+#include <cstdio>
+#include <exception>
+#include <sstream>
+
+#include <solv/pool.h>
+#include <solv/repo.h>
+#include <solv/repo_solv.h>
+#include <solv/repo_write.h>
+
+#include "solv-cpp/repo.hpp"
+
+namespace mamba::solv
+{
+    /****************************************
+     *  Implementation of ConstObjRepoView  *
+     ****************************************/
+
+    ObjRepoViewConst::ObjRepoViewConst(const ::Repo* repo) noexcept
+        : m_repo{ repo }
+    {
+    }
+
+    auto ObjRepoViewConst::raw() const -> const ::Repo*
+    {
+        return m_repo;
+    }
+
+    auto ObjRepoViewConst::id() const -> RepoId
+    {
+        // No function exists to get a repo id
+        const ::Pool* pool = raw()->pool;
+        const ::Repo* repo = nullptr;
+        int repoid = 0;
+        FOR_REPOS(repoid, repo)
```

Comment:
Could we cache the id after having done the search once? If the id doesnt change for the same value that might avoid having to loop through the ids each time we call this function that looks like a cheap getter from outside...

Reply:
I thought the same, and that we could even pass it a construction time when we have it available.

The list of repo is likely to be small (less than 10) and this function should not be called much so I did not bother to (for the sake of keeping a simple wrapper around a pointer).